cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/AbstractSchema.scala

```diff
@@ -84,7 +84,14 @@ object Field {
         RecordField(name, recordField.schema, recordField.isOptional)
       case DataType.Bag =>
         val bagField = field.asInstanceOf[BagField]
-        BagField(name, bagField.innerType, bagField.isOptional)
+        bagField.innerType match {
```

Comment:
If we modify the logic here, in `as` function of a field, it might cause some problem for other queries, because you are assuming that `Bag` typed field can only be called in the sub-statement of `unnest`. For example, if we have a query using `Bag` typed field in the `Select` sentence of a `Lookup` sub-statement, this hard-coded type-transferring may cause some problem. I think, we may to need find a way to handle this `as` logic specific to `unnest` cases.